app/scripts/directives/serviceIntegration.js

```diff
@@ -55,18 +56,36 @@
     ctrl.$onInit = function() {
       var context = {namespace: ctrl.consumerService.metadata.namespace};
 
+      //setup watch on config maps, looks for the consumer service's configmap to get its service name
+      watches.push(DataService.watch(configMapPreferredVersion, context, function(configMapsData) {
+        var consumerServiceConfigMap = _.find(configMapsData["_data"], function(configMap) {
+          return _.get(configMap, "metadata.labels.serviceInstanceID") === ctrl.consumerService.spec.externalID;
+        });
+        $scope.consumerServiceName = _.get(consumerServiceConfigMap, 'metadata.labels.serviceName', {});
+      }));
+
       //setup watch on service instances, looking for an instance which provides this integration
       watches.push(DataService.watch(instancePreferredVersion, context, function(serviceInstancesData) {
         var data = serviceInstancesData.by('metadata.name');
-        $scope.serviceInstance = _.find(data, function(serviceInstance) {
+        $scope.providerServiceInstance = _.find(data, function(serviceInstance) {
           var clusterServiceClassExternalName = _.get(serviceInstance, "spec.clusterServiceClassExternalName");
           return (clusterServiceClassExternalName === ctrl.integration.spec.externalName);
         });
+
+        // get the service name of the provider service from its configmap.
+        if ($scope.providerServiceInstance) {
+          DataService.list(configMapPreferredVersion, context).then(function(configMapData) {
```

Comment:
I think this can actually be put under the watch of configmaps so we don't need to list it here again since the provider service instance is in the scope already. What's the reason why we do this line here `var data = configmaps.by('metadata.name');`?

Reply:
It just seems to be the way they access `_data`